src/Components/EnvironmentPicker/EnvironmentPicker.tsx

```diff
@@ -593,74 +596,175 @@ const EnvironmentPicker = (props: EnvironmentPickerProps) => {
                 <div className="cb-environment-picker-container">
                     <FontIcon iconName={'Database'} />
                     <span className="cb-environment-picker-container-title">
-                        {displayNameForContainer(selectedContainerUrl)}
+                        {displayTextForContainer(
+                            containerItems.selectedContainer
+                        )}
                     </span>
                 </div>
             )}
 
             <Dialog
-                hidden={isDialogHidden}
+                hidden={false}
+                styles={{
+                    root: { display: !isDialogHidden ? 'flex' : 'none' }
```

Comment:
Please see my other comment above about firstTimeVisible value for more explanation. 

The issue is when Dialog fluent component is open and closed the content of it [is being unmounted](https://github.com/microsoft/fluentui/issues/11529) (all the ResourcePicker children components) - so it was a problem since we dont want to unmount and mount and refetch the resources again and want to keep the same loading process going while we close and open the dialog by clicking edit icon. That is why I applied the suggested solution for it - just always visible but changing the css. This also answers your other question below.

Reply:
I understand the reasoning, that makes sense. But we still should not do this. That doesn't fix the accessibility issue. Let's think about alternative solutions to that problem rather than doing this.

Reply:
@darsney you have any creative ideas here?